src/modules/ProductLicense.rb

```diff
@@ -1144,12 +1144,24 @@ def AskLicenseAgreement(src_id, dir, patterns, action, enable_back, base_product
       end
 
       licenses_ref = arg_ref(licenses)
-      DisplayLicenseDialog(
+
+      label = Pkg::SourceGeneralData(src_id)["name"]
+      title = _("License Agreement")
+
+      if !label.empty?
+        title = _("%s License Agreement") % label
+      end
+
+      eula_url = Pkg::SourceGeneralData(src_id)["url"]
+      Yast::ProductLicense.license_file_print = eula_url
```

Comment:
Since I actually tested my code I found out that the license filename gets appended elsewhere*. So the repo URL is enough here. See screenshot.

* actually in update_license_location() line 1602
![addonlicense_fixed](https://cloud.githubusercontent.com/assets/336868/6467747/7089fb82-c1d0-11e4-8bd2-fe9097c24c78.png)